pandas/tests/groupby/test_groupby.py

```diff
@@ -2709,6 +2709,45 @@ def test_single_element_list_grouping():
     assert result == expected
 
 
+def test_single_element_list_level_grouping_deprecation():
```

Comment:
can you parametrize these tests using `@pytest.mark.parametrize`. For the MultiIndex, I'd recommend a Boolean `multiindex` parameter and then use `if multiindex: df = df.set_index(['a', 'b'])`.

Reply:
Absolutely, I'll look into it!

Reply:
Should be resolved now :)